### ISSUE TITLE
In-app updates does not match the documentation behavior.

### ISSUE THREAD

**Issue by llDavidll** [applivery/applivery-android-sdk#39]
Hi,

I just started using the Applivery sdk for Android. I followed the documentation :
- I call the init method in the `onCreate` method of a custom Application
`Applivery.init(this, BuildConfig.APP_TOKEN, false);`
- I have a login screen, which is sometime "not stable" : when your are already logged in, your are automatically redirected to the home screen. So I am not calling any other method until I get to the home screen. Then I call :
`Applivery.checkForUpdates()`

The documentation states :

> Once initialized the SDK and once your App is stable in the Home Screen you have to call proactivelly the following method in order to check for new updates:
`Applivery.checkForUpdates()`

So I expect the sdk to check for updates and display the update popup on the home screen only, since the `setCheckForUpdatesBackground` parameter is false by default.

My test project is using public distribution, with suggested updates (not mandatory).

The behavior I'm getting is quite different :
- The update popup is displayed right after the call to the init method, without any call to checkForUpdates in my code.
- When I do not click the "Later" button of the first popup (which is tricky since my timer launches the home activity quickly), I never see the update dialog again until I kill the app.

I did some research in debug, and found the following :

The sdk will always check for updates on the first init call. See `AppConfigChecker.shouldCheckAppConfigForUpdate` : `AppliverySdk.isSdkFirstTime()` always returns true at this point.
The checkForUpdates method is triggered by the `AppliveryActivityLifecycleCallbacks` : the first activity that starts triggers a check for updates call, and it triggers the update popup since `shouldCheckAppConfigForUpdate` always returns true at launch.

Another bug seems to come from `SuggestedUpdateViewImpl`. Is contains a static `isVisible` boolean to avoid displaying two dialogs. It is set to false when the user clicks one of the buttons, but the dialog is cancelable (using the back button of clicking outside the dialog). So if you close the dialog this way, the boolean is never set to false, and the only way to display a dialog again is to kill the app. I see two quick ways to fix this :
- Set the dialog not cancelable
- Add a dismiss listener to the dialog to set this boolean to false

### Workaround
If someone is interested in a workaround, here is what I did :

Application's onCreate :
```
Applivery.init(this, BuildConfig.APP_TOKEN, false);
AppliverySdk.setSdkFirstTimeFalse()
AppliverySdk.setSdkRestartedFalse()
```

Home screen's onCreate :
```
Applivery.checkForUpdates()
```

With this, the init call will not trigger the update dialog, but the home screen will do. If you use the dialog buttons to close it : it will display the dialog again when you re-open your home screen. If you dismiss the dialog another way : you have to kill the app to make it work again.

**Comment by CesarTrigo** [comment]
Hi @llDavidll,
Thank you for the accurate feedback. The team is looking into i.
Regards,
C.

**Comment by mrebollob** [comment]
Hi @llDavidll! Thanks for the feedback. We are working on fix this bug for the next release.
You could follow the updates in this PR:
https://github.com/applivery/applivery-android-sdk/pull/41
https://github.com/applivery/applivery-android-sdk/pull/42

**Comment by llDavidll** [comment]
Thank you for fixing this that fast, any idea on the next release date ?

**Comment by CesarTrigo** [comment]
Hi @llDavidll ,

We have just released a testing version solving this and other small bugs. It has been approved by our QA team but before releasing it, it would be great to have your confirmation that it works as expected.

The testing package has been published in [Bintray](https://bintray.com/applivery/applivery-android-snapshots/applivery-android-sdk/3.0.2.1-Snapshot).

You can install it by adding the following repository:
```
repositories {
    maven {
        url  "https://dl.bintray.com/applivery/applivery-android-snapshots" 
    }
}
```

 and then the following dependency to your App Gradle:
`compile 'com.applivery:applivery-sdk:3.0.2.1-Snapshot'`

We look forward to your confirmation.
Thank you,
C.

**Comment by llDavidll** [comment]
Hi,

Can you give me any updates on v3.1 ? I cannot deploy AppLivery SDK in my apps untils the two other issues are solved, but i'm paying for it since this month.
I was planning on deploying it at the end of september, so I would really like to have an idea on when the fixes will be available (next week, next month...).

**Comment by CesarTrigo** [comment]
Hi @llDavidll 
We expect to release the new version by the end of next week. Most probably Sept 25th/26th.
That release will for sure include issues #39 and #44. #45 is still to be defined and will keep you posted.

Regards,
C.

**Comment by CesarTrigo** [comment]
Released in [v3.0.3](https://github.com/applivery/applivery-android-sdk/releases/tag/3.0.3)
